api/app/guestRegistrations/api.py

```diff
@@ -0,0 +1,307 @@
+            current_user = db.session.query(AppUser).filter(AppUser.id == user_id).first()
+
+            registration_form = db.session.query(RegistrationForm).filter(
+                RegistrationForm.id == args['registration_form_id']).first()
+
+            if not registration_form:
+                return errors.REGISTRATION_FORM_NOT_FOUND
+
+            registration = GuestRegistration(
+                registration_form_id=args['registration_form_id'],
+                user_id=user_id,
+                confirmed=True,
+                created_at=date.today(),
+                confirmation_email_sent_at=date.today()
+            )
+
+            db.session.add(registration)
+            db.session.commit()
+
+            event_name = db.session.query(Event).filter(Event.id == registration_form.event_id).first().name
+            for answer_args in args['answers']:
+                if db.session.query(RegistrationQuestion).filter(
+                        RegistrationQuestion.id == answer_args['registration_question_id']).first():
+                    answer = GuestRegistrationAnswer(guest_registration_id=registration.id,
+                                                     registration_question_id=answer_args['registration_question_id'],
+                                                     value=answer_args['value'])
+                    db.session.add(answer)
+            db.session.commit()
+
+            registration_answers = db.session.query(GuestRegistrationAnswer).filter(
+                GuestRegistrationAnswer.guest_registration_id == registration.id).all()
+            registration_questions = db.session.query(RegistrationQuestion).filter(
+                RegistrationQuestion.registration_form_id == args['registration_form_id']).all()
+
+            self.send_confirmation(current_user, registration_questions, registration_answers, registration.confirmed,
+                                   event_name)
+
+            return registration, 201  # 201 is 'CREATED' status code
+        except SQLAlchemyError as e:
+            LOGGER.error("Database error encountered: {}".format(e))
+            return errors.DB_NOT_AVAILABLE
+        except Exception as e:
+            LOGGER.error("Encountered unknown error: {}".format(traceback.format_exc()))
+            return errors.DB_NOT_AVAILABLE
+
+
+    @auth_required
+    def put(self):
+        # Update an existing response for the logged-in user.
+        req_parser = reqparse.RequestParser()
+        args = self.req_parser.parse_args()
+        try:
+            user_id = verify_token(request.headers.get('Authorization'))['id']
+            registration = db.session.query(GuestRegistration).filter(
+                GuestRegistration.id == args['guest_registration_id']).first()
+            if registration is None:
+                return 'Registration not found', 404
+            registration.registration_form_id = args['registration_form_id']
```

Comment:
This is the put, so updating the existing registration questions,
If you somehow manager to execute put on front end instead of the post before a registration exists i presume this shouldn't be done, however i doubt this will be the case, according to how the front end is calling the api, n

Reply:
Okay cool.